### ISSUE TITLE
Uncertainties increase with resolution

### ISSUE THREAD

**Issue by GilesStrong** [GilesStrong/tomopt#45]
# Problem

Whilst investigating #43 I found that the gradient of the scatter-location uncertainties were consistently positive, i.e. as the resolution increases, the uncertainty increases. This feeds into the x0 prediction uncertainties, which tend to have positive gradients (but occasionally can be negative).

I'd have thought that increasing the resolution should always decrease the uncertainty on scatter location.

# Checks

In `ScatterBatch`
```python
# loc uncertainty
dloc_dres = torch.stack([jacobian(self._loc, l.resolution).sum((-1, -2)) for l in dets], dim=1)
self._loc_unc = torch.sqrt((dloc_dres.pow(2) * res2).sum(1))
print('scatter dloc/dres', jacobian(self._loc, dets[0].resolution, create_graph=True, allow_unused=True).sum((-1, -2))[:10])
print('scatter dlocunc/dres', jacobian(self._loc_unc, dets[0].resolution, create_graph=True, allow_unused=True).sum((-1, -2))[:10])
```

produces:
<img width="634" alt="Screenshot 2021-06-23 at 13 06 02" src="https://user-images.githubusercontent.com/18284037/123086519-6cd47300-d41b-11eb-98ca-4ff5e69d2ea3.png">

So resolution can shift the scatter location in either direction of each dimension, but the shift always increases as resolution increases. Similar checks on dtheta, dxy, and theta_in uncertainties all showed that increasing resolution increases uncertainty.

I assume that there is something wrong with the way we calculate the uncertainty:
unc_x = sqrt(sum_i [(res_i*dx/dres_i)^2]), where i sums over the 4 detector layers
Does this formula appear to be correct?
